core/src/main/java/cucumber/runtime/formatter/PrettyFormatter.java

```diff
@@ -213,16 +213,22 @@ String formatStepText(String keyword, String stepText, Format textFormat, Format
         for (Argument argument : arguments) {
             // can be null if the argument is missing.
             if (argument.getOffset() != null) {
-                String text = stepText.substring(textStart, argument.getOffset());
-                result.append(textFormat.text(text));
+                int textEnd = argument.getOffset();
+                if (textStart > textEnd ) { // textStart can be > textEnd with nested capture groups
+                    continue;
+                } else {
+                    String text = stepText.substring(textStart, textEnd);
+                    result.append(textFormat.text(text));
+                }
+
             }
             // val can be null if the argument isn't there, for example @And("(it )?has something")
             if (argument.getVal() != null) {
                 result.append(argFormat.text(argument.getVal()));
                 textStart = argument.getOffset() + argument.getVal().length();
             }
         }
-        if (textStart != stepText.length()) {
+        if (textStart != stepText.length() && textStart >= 0 && textStart < stepText.length()) { // textStart can be < 0 with nested capture groups
```

Comment:
Can you explain (or even better write  a test) for the scenario where `textStart` becomes less then zero or longer then the stepText now that we are skipping nested arguments?